### PR TITLE
META-30: Setup GraphQL Server

### PR DESCRIPTION
**Pull request by pczern**
## Description
sets up GraphQL server with Mercurius (GraphQL middleware) and Fastify (server)

## How I implemented
* `concurrently` runs two scripts simultaneously so we can run `tsc -w` (watch mode compiles typescript files to js in `/dist` folder on every file change) and nodemon (restarts server on file changes)
* GraphQL server uses two example Git Repositories coded as array
* Comments with Links to helpful GraphQL resources

## Other 
* Fastify allows faster requests than Expressjs
* Mercurius was used because Apollo Server doesn't support Fastify officially, middlewares are not guranteed to be maintained, they are maintained by community, only ExpressJs is officially supported, new versions often introduce breaking changes, and not everything is MIT licensed like Apollo Router!
* Go to `http://localhost:3000/graphiql` and run to execute your first query
```gql
{
   gitRepositories {
    name
  }
}
```

Related to issue #BACK-18

### CODE REVIEW

**Review by cnk1dev — CHANGES_REQUESTED** [review]
Please also move graphql/, listen.ts and server.ts into a src/ folder for a cleaner project structure. Besides the other 2 issues I pointed out, everything looks good.

File: server.ts
[the remark this answers is not in the document]
Comment (pczern) [reply server.ts:1]: done

**Comment by pczern** [comment]
> Please also move graphql/, listen.ts and server.ts into a src/ folder for a cleaner project structure. Besides the other 2 issues I pointed out, everything looks good.

files moved to ./src folder @cnk1dev

**Review by cnk1dev — APPROVED** [review]
Changes requested were addressed, looks good now

**Pull request merged**